config/entry.go

```diff
@@ -192,3 +202,40 @@ func (e *Entry) convertEnvVar(str, key string) (any, error) {
 
 	return nil, nil
 }
+
+func (e *Entry) validateRequired(kind reflect.Kind, key string) error {
```

Comment:
I think we should only check is the value is `nil`. There are some legitimate use-cases where you would want to use `0` or an empty string as a value.

The main benefit of adding a required flag is to notify developers when they forgot to add an entry in their config file. We should limit ourselves to this in my opinion.

Reply:
Only checking if the value is `nil` won't cover someone making **incorrect** configuration, for example in the case of containerizing a project, where (e.g. with Kubernetes) env variables are injected directly in the container and then used in the config. 

If someone explicitly wants to use empty or 0, they can choose not to make it 'required'. 

Then again, I would rather like support for the validation RuleSet to be supported for the configuration so devs can decide for themselves what validation they want. What is your opinion on that?

Reply:
I did some tests and you can indeed omit the complete value from the config. Having Required without any additional checks (empty or `nil`) would already benefit indeed.

Let me know what your ideas are on a clean way the handle adding a RuleSet to the config. Are you going to do that in an issue, or in Discord?

Reply:
Functionality removed ✔️

Reply:
I created a [Github discussion](https://github.com/go-goyave/goyave/discussions/212) on this topic. We'll have to think about it more.